document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -265,8 +265,15 @@ private BasicDBObject prepareDocument(Key key, Document document, long now)
   /** Updates auto-field lastUpdatedTime when sub doc is updated */
   @Override
   public boolean updateSubDoc(Key key, String subDocPath, Document subDocument) {
+    String jsonString = subDocument.toJson();
     try {
-      BasicDBObject dbObject = getSanitizedObject(subDocument);
+      JsonNode jsonNode = MAPPER.readTree(jsonString);
```

Comment:
What failed when using the method version? Do we have different sanitization in different places?

Reply:
This seems concerning to me - what's different? which one is correct?

Reply:
Missed the subdoc path. Fixed it.